### PR TITLE
interfaces: add thumbnailer interface

### PR DESCRIPTION
**Pull request by fkaleo**

### CODE REVIEW

File: interfaces/builtin/thumbnailer_test.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+	iface interfaces.Interface
+	slot  *interfaces.Slot
+	plug  *interfaces.Plug
+}
+
+var _ = Suite(&BluezInterfaceSuite{
+	iface: &builtin.BluezInterface{},
+	slot: &interfaces.Slot{
+		SlotInfo: &snap.SlotInfo{
+			Snap:      &snap.Info{SuggestedName: "bluez"},
```
Comment (zyga) [inline interfaces/builtin/thumbnailer_test.go]: This is copy pasted, it would be nice to at least replace those strings.

**Comment by zyga** [comment]
What is the indented use case of this interface? It seems to be used on classic (because of the `peer=unconfined` label. If that is the case did you consider adding this interface to the list of implicit slots added on classic (see `snap/implicit.go`)

**Comment by zyga** [comment]
Please check the README.md for some hints on how to run unit tests. You should also re-format the files you changed with `go fmt` so that they match the coding standard used by snapd.

**Comment by fkaleo** [comment]
Sorry about that: most of my work was missing from the PR due to the missed commit.
About the snap/implicit.go, thank you I was not aware of that. Is there some documentation of what it means?

File: interfaces/builtin/thumbnailer.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+</policy>
+<policy context="default">
+    <deny send_destination="com.canonical.Thumbnailer"/>
+</policy>
+`)
+
+type ThumbnailerInterface struct{}
+
+func (iface *ThumbnailerInterface) Name() string {
+	return "thumbnailer"
```
Comment (morphis) [inline interfaces/builtin/thumbnailer.go]: Can you document this interface in https://github.com/snapcore/snapd/blob/master/docs/interfaces.md?

Also a manual test case to verify the interface is working in https://github.com/snapcore/snapd/blob/master/integration-tests/manual-tests.md or a spread test would be required.

File: interfaces/builtin/thumbnailer.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+# Allow traffic to/from org.freedesktop.DBus for thumbnailer service
+dbus (receive, send)
+    bus=session
+    path=/com/canonical/Thumbnailer{,/**}
+    interface=org.freedesktop.DBus.*,
+`)
+
+var thumbnailerServiceConnectedPlugAppArmor = []byte(`
+# Description: Allow accessing the thumbnailer service.
+# Usage: reserved
```
Comment (morphis) [inline interfaces/builtin/thumbnailer.go]: Why is the plug reserved as well?

Comment (fkaleo) [inline interfaces/builtin/thumbnailer.go]: Removed 'reserved'

File: interfaces/builtin/thumbnailer.go (hunk 2 above)
Comment (fkaleo) [inline interfaces/builtin/thumbnailer.go]: I added a spread test that queries the thumbnailer service over dbus. Are the spread tests run in the PR?

File: interfaces/builtin/thumbnailer.go [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    <allow own="com.canonical.Thumbnailer"/>
+    <allow send_destination="com.canonical.Thumbnailer"/>
+    <allow send_interface="com.canonical.Thumbnailer"/>
+    <allow send_interface="org.freedesktop.DBus.ObjectManager"/>
+    <allow send_interface="org.freedesktop.DBus.Properties"/>
+</policy>
+<policy context="default">
+    <deny send_destination="com.canonical.Thumbnailer"/>
+</policy>
+`)
```
Comment (jhenstridge) [inline interfaces/builtin/thumbnailer.go]: I'm a snappy novice, but this looks like the kind of configuration you'd use to expose a service on the system bus.

Thumbnailer is a user service that runs on the session bus and serve requests from processes with the same UID.  It definitely isn't intended to be a system service running as root exposed on the system bus.

Comment (morphis) [inline interfaces/builtin/thumbnailer.go]: It is. @fkaleo can you clarify what kind of service the thumbnailer is?

**Comment by morphis** [comment]
@fkaleo: Can you look at the failing tests?

File: interfaces/builtin/thumbnailer.go (hunk 4 above)
Comment (fkaleo) [inline interfaces/builtin/thumbnailer.go]: Indeed it sounds like I did it all wrong. Despair.
@jhenstridge are we supposed to have a single instance of the service per user session or is it ok to bundle the thumbnailer service as part of the app and spawn one per app?

**Comment by fkaleo** [comment]
Just a note: once this interface is good to go, we will be missing:
- a thumbnailer-service snap (or perhaps another bigger snaps that includes all our Ubuntu Personal services
- a fix for apparmor's bug https://bugs.launchpad.net/snappy/+bug/1620635

**Comment by niemeyer** [comment]
What's the status here?

**Comment by niemeyer** [comment]
I see activity, but a question 10 days ago that went unanswered, missing files, conflicts, failing tests.

It doesn't look like this should be in review yet. Closing, please reopen when it's golden.

**Pull request closed without merging**